### PR TITLE
App navigation via unified search

### PR DESCRIPTION
**Pull request by nickvergessen**
Bildschirmfoto von 2020-08-04 16-23-12

Branch targets #22099

### CODE REVIEW

**Comment by skjnldsv** [comment]
Pls rebase :p

**Comment by jancborchardt** [comment]
As discussed:

>  We could possibly add "Help" in results (referring to documentation). E.g. macOS also has these nice "Help" results when searching inside of an app via the app menu.

This should be sorted all the way down, below settings. Cause ideally settings already gives you the answer, and then "Help" is the last error prevention. :)

**Comment by nickvergessen** [comment]
Rebased with order adjustments

**Review by skjnldsv — APPROVED** [review]

File: apps/settings/lib/Search/AppSearch.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+			if (strpos($query->getRoute(), $entry['id'] . '.') === 0) {
+				// Skip the current app, unlikely this is intended
+				continue;
+			}
+
+			$result[] = new SearchResultEntry(
+				'',
+				$entry['name'],
+				'',
```
Comment (skjnldsv) [inline apps/settings/lib/Search/AppSearch.php:93]: Btw, is there no other way for an API to have empty strings like that? Is that good practices?

File: apps/settings/lib/Search/AppSearch.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	 */
+	public function getName(): string {
+		return $this->l->t('Apps');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getOrder(string $route, array $routeParameters): int {
+		return -1;
```
Comment (skjnldsv) [inline apps/settings/lib/Search/AppSearch.php:66]: Jan said it should always be the very top, so maybe something crazy like -50 ?

Comment (skjnldsv) [reply apps/settings/lib/Search/AppSearch.php:66]: I'll do it in a followup, let get this in

**Review by jancborchardt — CHANGES_REQUESTED** [review]
Looks good! :+1: to @skjnldsv’s feedback, and also I found one entry which doesn’t really make sense – there’s possibly more which we might need to filter?
image

**Review by jancborchardt — APPROVED** [review]
My feedback can also be fixed in a follow-up tho.

**Comment by nickvergessen** [comment]
> there’s possibly more which we might need to filter?

Well user status is registering a navigation link which is highjacked by JS. I'm not aware we have more of this, but now fixed by ignoring empty link items

Fix both issues.

**Review by MorrisJobke — APPROVED** [review]
How nice is this?!? :) I love it already 👍

**Pull request merged**